### DISCUSSION TITLE
Compositor-driven portal configuration

### DISCUSSION THREAD

**Discussion by mahkoh** [flatpak/xdg-desktop-portal#1293] [Ideas]
Portals and their mapping to compositors are currently configured via files. Some of these files require root permissions to create.

This is a problem for compositors that are not installed via the system package manager and don't have any natural way to create such files.

Would it instead be possible to create a wayland protocol that xdg-desktop-portal can use to retrieve this information from the compositor it is running under? This would be a more direct approach than using the XDG_CURRENT_DESKTOP and would allow compositors to fully integrate with xdg-desktop-portal without having to modify the file system.

**Comment by Mikenux** [comment]
I'm not the right person for this topic, but I think it would be helpful if you explain the use case, possibly giving examples.

**Reply by mahkoh** [reply]
The usecase is compositors that are not installed via the system package manager.

I'm currently preparing the first stable release of my compositor and the recommended way to install it will be via `cargo install <compositor>`. This will place the executable somewhere under $HOME. Requiring the user to manually move some configuration files to specific locations is both error prone and annoying. Having flatpak and xdg-desktop-portal retrieve this information from the compositor would remove this manual step.
